test/integration/test_phone_number_lookup.py

```diff
@@ -0,0 +1,168 @@
+from bandwidth.model.create_lookup_response import CreateLookupResponse
+from bandwidth.model.lookup_status import LookupStatus
+from bandwidth.model.lookup_result import LookupResult
+from bandwidth.model.lookup_status_enum import LookupStatusEnum
+from bandwidth.model.tn_lookup_request_error import TnLookupRequestError
+from bandwidth.exceptions import UnauthorizedException, ForbiddenException
+
+
+class TestPhoneNumberLookupIntegration(unittest.TestCase):
+    """Phone Number Lookup API integration test"""
+
+    
+    def setUp(self):
+        configuration = bandwidth.Configuration(
+            username = os.environ['BW_USERNAME'],
+            password = os.environ['BW_PASSWORD']
+        )
+        api_client = bandwidth.ApiClient(configuration)
+        self.api_instance = phone_number_lookup_api.PhoneNumberLookupApi(api_client)
+        self.account_id = os.environ['BW_ACCOUNT_ID']
+
+    
+    def testSuccessfulPhoneNumberLookup(self):
+        """Test Phone Number Lookup API"""
+        lookup_request = LookupRequest(
+            tns=[
+                os.environ['BW_NUMBER'],
+                os.environ['VZW_NUMBER']
+                # os.environ['BW_INVALID_TN_LOOKUP_NUMBER']
+            ],
+        ) 
+
+        create_lookup_response: CreateLookupResponse = self.api_instance.create_lookup(self.account_id, lookup_request, _return_http_data_only=False)
+        self.assertEqual(create_lookup_response[1], 202)
+        self.assertIs(type(create_lookup_response[0].status), LookupStatusEnum)
+        self.assertEqual(create_lookup_response[0].status, LookupStatusEnum("IN_PROGRESS"))
+        self.assertIs(type(create_lookup_response[0].request_id), str)
+        self.assertRegex(create_lookup_response[0].request_id, r'^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$')
+        
+        get_lookup_status_response: LookupStatus = self.api_instance.get_lookup_status(self.account_id, create_lookup_response[0].request_id, _return_http_data_only=False)
+        self.assertEqual(get_lookup_status_response[1], 200)
+
+        # make the request again without _return_http_data_only=False to just get the LookupStatus model
+        get_lookup_status_response: LookupStatus = self.api_instance.get_lookup_status(self.account_id, create_lookup_response[0].request_id)
+        get_lookup_status_response_attempts = 1
+        while get_lookup_status_response.status != LookupStatusEnum('COMPLETE'): 
+            # Raise an error if it takes more than 5 requests to get COMPLETE status
+            if get_lookup_status_response_attempts == 5:
+                raise Exception(f'Took too long to get phone number lookup \'COMPLETE\' status. Aborting test after {get_lookup_status_response_attempts} attempts.')
+            time.sleep(2)
+
+            get_lookup_status_response: LookupStatus = self.api_instance.get_lookup_status(self.account_id, create_lookup_response[0].request_id)
+            get_lookup_status_response_attempts += 1
```

Comment:
💯 for adding check for `COMPLETE` status.

At the moment the above test code is not reused but still suggest to abstract it to it's own function giving expressive name for what it intends to do. This will reduce cognitive load next developer supporting it.

Reply:
Created a `pollLookupStatus` function that returns the LookupResult to handle this